util/src/main/scala/org/ensime/util/file.scala

```diff
@@ -115,6 +115,9 @@ package object file {
         case t: Throwable => file.getAbsoluteFile
       }
 
+    def descendsFrom(parent: File): Boolean =
+      file.getParent.startsWith(parent.getAbsolutePath)
```

Comment:
The only VFS based approach I can find would be transforming each of the module roots into a `FileObject` then using `getChild`. I haven't tested this yet, so please let me know if there's another  way of performing this check that you had in mind.